### PR TITLE
Ds photogram industrial 2

### PR DESCRIPTION
**Pull request by dylansarikas**
Pull Request for Photogram Industrial 2

### CODE REVIEW

**Review by CodeX-y — APPROVED** [review]

**Comment by CodeX-y** [comment]
Looks good!

**Review by APatel-AI — APPROVED** [review]
Looks good and Well structured

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Foreign Keys
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
+
+  validates :fan_id, uniqueness: { scope: :photo_id, message: "has already liked this post" }
```
Comment (thierrychau) [inline app/models/like.rb:25]: Good job!

File: app/models/user.rb [hunk 2]
```diff
@@ -27,4 +28,18 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
 
   has_many :own_photos, class_name: "Photo", foreign_key: "owner_id"
+  has_many :comments, foreign_key: "author_id"
+  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest", dependent: :destroy
```
Comment (thierrychau) [inline app/models/user.rb:32]: Great job!

**Review by thierrychau — APPROVED** [review]
Great job, Dylan. Good to merge!

**Review by ghost — COMMENTED** [review]
Wow, great work Dylan!

**Pull request merged**